micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemoryMetrics.java

```diff
@@ -100,6 +98,58 @@ public void bindTo(MeterRegistry registry) {
                 .baseUnit(BaseUnits.BYTES)
                 .register(registry);
         }
+
+        //The used and committed size of the returned memory usage is the sum of those values of all heap/non-heap memory pools
+        // whereas the max size of the returned memory usage represents the setting of the non-heap/non-heap memory which may
+        // not be the sum of those of all non-heap memory pools. If the setting is missing for heap/non-heap memory pools these values come as -1.
+
+        MemoryMXBean memoryMXBean = ManagementFactory.getMemoryMXBean();
+
+        Iterable<Tag> tagsWithId = Tags.concat(tags,"id", "total", "area", "heap");
+
+        Gauge.builder("jvm.memory.used", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getUsed, "heap"))
+                .tags(tagsWithId)
+                .description("The amount of used memory")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.max", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getMax, "heap"))
+                .tags(tagsWithId)
+                .description("The maximum amount of memory in bytes that can be used for memory management")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.committed", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getCommitted, "heap"))
+                .tags(tagsWithId)
+                .description("The amount of memory in bytes that is committed for the Java virtual machine to use")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.heap_used_percent", memoryMXBean, (mem) -> getHeapUsagePercent(mem))
```

Comment:
we should only use `.` and not `_` in meter names and tag keys.

Reply:
Changed meter name to use `.` instead of `_` .